exp/api/v1beta1/azuremanagedmachinepool_webhook.go

```diff
@@ -319,6 +336,21 @@ func (m *AzureManagedMachinePool) validateName() error {
 	return nil
 }
 
+func (m *AzureManagedMachinePool) validateNodeLabels() error {
```

Comment:
I don't think it's possible to do this validation in the kubebuilder API type spec (e.g., using the `Pattern=` interface to define a required regular expression match) because the API type is a `map[string]string`.